doc/modules/mod_event_pusher_push.md

```diff
@@ -141,23 +72,47 @@ Please note the value of `virtual_pubsub_hosts` option.
 
 ## Plugin module
 
-A plugin module handles dynamic configuration of push notifications. It implements `mod_event_pusher_push_plugin` behaviour which
-requires two callbacks:
+You can also control the format of the "sender" of the push notification (which ultimately becomes
+the title of push notification) and filter which messages will trigger the notification.
+In order to achieve that, you need to create a plugin module that implements the
+`mod_event_pusher_push_plugin` behaviour and enable this plugin in the `plugin_module` section as
+above.
 
-* `should_publish/3` - callback used for filtering push notifications. A push notification is triggered for a given a message only if this
-callback returns `true`.
+A plugin module handles the dynamic configuration of push notifications. 
+It contains the filtering and custom logic for notifying about messages.
 
-```
--spec should_publish(From :: ejabberd:jid(), To :: ejabberd:jid(), Packet :: jlib:xmlel()) -> boolean().
-```
+Two plugin implementations are provided.
+They offer different behaviour considering unacknowledged messages when using [Stream Management][XEP-0198]:
 
-* `publish_notification/5` - does the actual push.
-  By default it pushes to the registered pubsub nodes (or executes the internal hook in case of a publish to a virtual domain).
 
-```
--spec publish_notification(Acc :: mongooseim_acc:t(), From :: jid:jid(),
-                           To :: jid:jid(), Packet :: exml:element(),
-                           Services :: [mod_event_pusher_push:publish_service()]) -> mongooseim_acc:t().
-```
+* `mod_event_pusher_push_plugin_defaults`, which implements an older behaviour. It does not notify
+  the user of unacknowledged messages immediately after detecting a lost connection to the user.
+* `mod_event_pusher_push_plugin_enhanced`, which pushes notifications as soon as the server detects
+  that the client has disconnected and waits for stream resumption. The immediate notification
+  prevents the unneeded suspension of the client's application, if there are any not yet
+  unacknowledged messages. This allows to create more power efficient mobile applications.
+
+In order for the enhanced plugin to work, each device (an entity that may receive push
+notifications) should be uniquely identified. The only correct way to identify a device from the
+XMPP standpoint is to use the data provided with the [enable stanza][enabling]. Because of that
+each device should (re)enable the push notifications at the beginning of each and every connection.
+
+### Custom plugins
+
+A custom module implementing the optional callbacks of `mod_event_pusher_push_plugin`
+may be used as a plugin to change the default behaviour. In case of not implemented callbacks
+the defaults are used instead.
+
+### Unacknowledged message hook
```

Comment:
this part must be moved to `/docs/push_notifications/doc/developers-guide/Hooks-and-handlers.md`

Reply:
I think you meant to `hooks_descriptions.md`, which is the one that describes a few hooks. Moved there.

Translation logic, I think that the code is way clearer than any docs, so that exact point I wasn't planning to explain, unless somebody asks me to 🤷‍♂